python/pipeline/experiment.py

```diff
@@ -593,4 +593,35 @@ class AutoProcessing(dj.Manual):
     autosegment=false   :boolean       # segment somas in the first channel with default method
     """
 
+@schema
+class Projector(dj.Lookup):
+    definition = """
+    # projector specifications
+    projector_id        : tinyint                               # projector id
+    refresh_rate        : tinyint                               # refresh rate in Hz
```

Comment:
can the refresh rate not be a float?
and if the refresh rate is different is it still the same projector?, bit odd to see the refresh rate (which I thought would be a dependent attribute of the projector as primary key)

Reply:
refresh rate is already in int? not a float?

Refresh rate can be _adjusted_ on the hardware level settings. So by specifying it as a primary key, we ensure that the refresh rate that was set by the project (physical level) must match with which projector AND which refresh setting (software level).

Reply:
I was asking whether you are sure refresh rate will never need to be a float

Reply:
that still seems like something that should be documented but not as part of the primary key. It seems class Projector is doing too many things, both recording different types of projector (whose key will be projector id and should have projector properties experimenters think are improtant, like maker of the projector, when we acquired it, how rgb inputs map to projected colors etc) and different configurations of that projector (which right now is just refresh_rate but may need more stuff to be saved and hopefully is projector independent), you could separate it into different classes. why do that, because if I have a table that depends on the projector but not its particular configuration I couldn't point to Projector right now becuase that will state that it depends on an specific refresh_rate too

Reply:
here's the theoretical reason why it seems odd: http://agiledata.org/essays/dataNormalization.html It't not normalized (it's only 1NF, not 2NF as non-dependent attributes do not depend on the refresh rate)

Reply:
oh. well i can change to `decimal` then

Reply:
refresh_rate should be a secondary attribute, not in primary key